pkg/client/devops/jenkins/request.go

```diff
@@ -310,22 +310,28 @@ func (r *Requester) Do(ar *APIRequest, responseStruct interface{}, options ...in
 	if fileUpload {
 		body := &bytes.Buffer{}
 		writer := multipart.NewWriter(body)
-		for _, file := range files {
-			fileData, err := os.Open(file)
+		uploadFunc := func(fileName string) error {
+			fileData, err := os.Open(fileName)
 			if err != nil {
 				Error.Println(err.Error())
-				return nil, err
+				return err
 			}
+			defer fileData.Close()
 
-			part, err := writer.CreateFormFile("file", filepath.Base(file))
+			part, err := writer.CreateFormFile("file", filepath.Base(fileName))
```

Comment:
I'm wondering if the variable `writer` can be a variable in this function. For instance:

```go
func uploadFunc(fileName string, writer multipart.Writer) error {}
```

Moving it be outside of `func (r *Requester) Do(ar *APIRequest, responseStruct interface{}, options ...interface{})` would be easier to add unit tests for it.

Reply:
Thanks for your suggestion. I submitted it again